src/cls/journal/cls_journal.cc

```diff
@@ -540,10 +540,12 @@ int journal_client_register(cls_method_context_t hctx, bufferlist *in,
                             bufferlist *out) {
   std::string id;
   bufferlist data;
+  std::set<cls::journal::Client> clients;
   try {
     bufferlist::iterator iter = in->begin();
     ::decode(id, iter);
     ::decode(data, iter);
+    ::decode(clients, iter);
```

Comment:
No need to pass in the clients (might be stale and it breaks compatibility with older clients).  This class method should just read the current client list keys and perform the logic.